hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/ReconUtils.java

```diff
@@ -131,9 +132,36 @@ public static boolean triggerAsyncNSSummaryRebuild(
       ReconNamespaceSummaryManager reconNamespaceSummaryManager,
       ReconOMMetadataManager omMetadataManager) {
     
+    // Check if a rebuild is already in progress
+    if (getNSSummaryRebuildState() == NSSummaryTask.RebuildState.RUNNING) {
+      log.info("NSSummary rebuild already in progress; skipping duplicate trigger.");
+      return false;
+    }
+    
     // Submit rebuild task to single thread executor for async execution
     NSSUMMARY_REBUILD_EXECUTOR.submit(() -> {
       try {
+        // Wait for OM tables to be initialized before proceeding
+        long maxWaitMillis = 300_000L; // 5 minutes
+        long waited = 0L;
+        long waitInterval = 1000L; // 1 second
+        
+        while (!omMetadataManager.isOmTablesInitialized() && waited < maxWaitMillis) {
```

Comment:
under what case its not initialized? I think should be initailized before calling and this wait might not be required.

Reply:
Changed as discussed!